resources/dev/scripts/telemetry.sh

```diff
@@ -0,0 +1,168 @@
+  echo "-----------------------------------------------------------------------------------------------------"
+	kubectl delete -f "${PROMETHEUS_OPERATOR_YAML}"
+	sleep 10
+
+	log_time "destroy-prometheus-operator"
+}
+
+function deploy-prometheus() {
+  echo ""
+	echo "Deploying prometheus"
+	echo "PROMETHEUS_RBAC_YAML: ${PROMETHEUS_RBAC_YAML}"
+	echo "PROMETHEUS_YAML: ${PROMETHEUS_YAML}"
+	echo "Namespace: $NAMESPACE"
+  echo "-----------------------------------------------------------------------------------------------------"
+	kubectl create -f "${PROMETHEUS_RBAC_YAML}" -n "${NAMESPACE}"
+
+	# create ClusterRole binding with the correct namespace
+	# NOTE: take care of indentation in the yaml if it needs to be updated
+	echo "
+  apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRoleBinding
+  metadata:
+    name: prometheus
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: ClusterRole
+    name: prometheus
+  subjects:
+    - kind: ServiceAccount
+      name: prometheus
+      namespace: ${NAMESPACE}" | kubectl create -n "${NAMESPACE}" -f -
+
+	sleep 10
+
+	kubectl create -f "${PROMETHEUS_YAML}" -n "${NAMESPACE}"
+
+	sleep 10
+	echo "Waiting for prometheus to be active (timeout 900s)..."
+	kubectl wait --for=condition=Ready pods -l  app.kubernetes.io/name=prometheus --timeout 900s -n "${NAMESPACE}"
+
+	log_time "deploy-prometheus"
+}
+
+function destroy-prometheus() {
+  echo ""
+	echo "Destroying prometheus"
+	echo "PROMETHEUS_RBAC_YAML: ${PROMETHEUS_RBAC_YAML}"
+	echo "PROMETHEUS_YAML: ${PROMETHEUS_YAML}"
+	echo "Namespace: $NAMESPACE"
+  echo "-----------------------------------------------------------------------------------------------------"
+	kubectl delete -f "${PROMETHEUS_YAML}" -n "${NAMESPACE}" || true
+	kubectl delete -f "${PROMETHEUS_RBAC_YAML}" -n "${NAMESPACE}" || true
+	kubectl delete clusterrolebindings prometheus -n "${NAMESPACE}" || true
+	sleep 5
+
+	log_time "destroy-prometheus"
+}
+
+function expose_prometheus() {
+  export POD_NAME=$(kubectl get pods -l "app.kubernetes.io/name=prometheus,app.kubernetes.io/instance=prometheus" -o jsonpath="{.items[0].metadata.name}" -n "${NAMESPACE}")
+  kubectl port-forward "${POD_NAME}" 9090 -n "${NAMESPACE}" &
+  echo "Prometheus is exposed from ${POD_NAME} to port 9090"
+}
+
+function unexpose_prometheus() {
+  export POD_NAME=$(kubectl get pods -l "app.kubernetes.io/name=prometheus,app.kubernetes.io/instance=prometheus" -o jsonpath="{.items[0].metadata.name}" -n "${NAMESPACE}")
+  export PID=$(ps aux | grep "port-forward ${POD_NAME}" | sed -n 2p | awk '{ print $2 }')
+  [[ -z "${PID}" ]] && echo "No Prometheus port-forward PID is found" && return 0
+
+  if [[ "${PID}" ]]; then
+    echo ""
+    echo "Un-exposing Prometheus: ${POD_NAME} for PID: ${PID}"
+    echo "-----------------------------------------------------------------------------------------------------"
+    kill "${PID}" &>/dev/null || true
+  fi
+}
+
+function deploy_grafana_tempo() {
```

Comment:
why not just add grafana tempo or a grafana stack as an option into the helm charts?

Reply:
We don't want so to deploy Grafana-stack as part of network deploy. Ideally we want to use grafana-agent only which should be configurable to export the metrics, traces and logs to a separate Grafana deployment.

Reply:
I couldn't get the telemetry support working with what `fullstack-cluster-setup` deploys. Probably missed something. I am going to work on it next week. Target it to be able to visualize one of our production Grafana dashboard

Reply:
Here is the ticket for details and reference: https://github.com/hashgraph/solo/issues/208